Sources/GoodSwiftUI/ReadableContentWidthModifier.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  ReadableContentWidthModifier.swift
+//
+//  GoodSwiftUI
+//  Created by Filip Šašala on 31/12/2023.
+//
+
+import SwiftUI
+
+private struct ReadableContentWidth: ViewModifier {
+
+    private let measureViewController = UIViewController()
+
+    @State private var orientation: UIDeviceOrientation = UIDevice.current.orientation
+
+    func body(content: Content) -> some View {
+        content
+            .frame(maxWidth: readableWidth(for: orientation))
+            .onReceive(NotificationCenter.default.publisher(for: UIDevice.orientationDidChangeNotification)) { _ in
+                orientation = UIDevice.current.orientation
+            }
+    }
+
+    private func readableWidth(for _: UIDeviceOrientation) -> CGFloat {
+        measureViewController.view.frame = UIScreen.main.bounds
```

Comment:
How does this work for the split view? It should depends on the windows you are currently using in the application not about UIScreen bounds.